qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/DeploymentClusteredTest.java

```diff
@@ -102,6 +103,49 @@ public void shouldDistributedDeploymentWhenStoppedBrokerIsRestarted() {
     clientRule.waitUntilDeploymentIsDone(processDefinitionKey);
   }
 
+  @Test
+  public void shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted() {
+    // given
+    final var deploymentPartitionLeader =
+        clusteringRule.getLeaderForPartition(Protocol.DEPLOYMENT_PARTITION).getNodeId();
+    final var secondPartitionLeader = clusteringRule.getLeaderForPartition(2).getNodeId();
+    final var adminServiceLeaderTwo =
+        clusteringRule.getBroker(secondPartitionLeader).getBrokerContext().getBrokerAdminService();
+    adminServiceLeaderTwo.pauseStreamProcessing();
+
+    // when
+    final DeploymentEvent deploymentEvent =
+        clientRule
+            .getClient()
+            .newDeployResourceCommand()
+            .addProcessModel(PROCESS, "process.bpmn")
+            .send()
+            .join();
+    final var processDefinitionKey = deploymentEvent.getKey();
+
+    assertThat(
+            RecordingExporter.deploymentDistributionRecords()
+                .withIntent(DeploymentDistributionIntent.COMPLETED)
+                .withPartitionId(3)
+                .findFirst())
+        .isPresent();
+
+    clusteringRule.stopBroker(deploymentPartitionLeader);
+    adminServiceLeaderTwo.resumeStreamProcessing();
```

Comment:
:wrench: We could move this part in the `given` test block. 

The deployment is part of building up the scenario. The `when` block (i.e. the action) is the restart of the broker.